plugins/catalog/src/types.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  EntityTypePickerProps,
+  UserListPickerProps,
+} from '@backstage/plugin-catalog-react';
+
+export type CatalogPageOptionsProps = {
```

Comment:
Should probably just call this `CatalogPluginOptions` and use as the input type. I'm thinking we also slim it back down to just the `createButtonTitle` before shipping, unless there are some other specific things you'd like to try out customizing.

As a convention I think it could make sense to call this file `options.ts`, define `<Plugin>PluginOptions` as the public input type, along with `PluginOptions` as the internal type, and `useCatalogPluginOptions = usePluginOptions<PluginOptions>` as an internal utility helper.

Reply:
I don't really get the second half of the comment. 
What is the purpose of input would supply here and how this helper will do help :) 

What we do only with it now is:

```
const { createButtonTitle } = usePluginOptions<CatalogPluginOptions>();
```

Reply:
Yep it would just be a utility to help avoid mistakes within the plugin and reduce the number of imports, so that line would instead be

```
const { createButtonTitle } = useCatalogPluginOptions();
```

Reply:
Incorporated